src/augeas.c

```diff
@@ -2020,6 +2021,39 @@ int aug_escape_name(augeas *aug, const char *in, char **out) {
     return result;
 }
 
+int aug_filter_file(struct augeas *aug, const char *file) {
+    int result = -1;
+
+    api_entry(aug);
+
+    bool found = false;
+
+    list_for_each(module, aug->modules) {
```

Comment:
With the changes to `transform_load` that I suggested, and since you now work with the transforms set up in `/augeas/load`, this loop could become something like
```
/* Setup basically copied from aug_load */
struct tree *meta = tree_child_cr(aug->origin, s_augeas);
struct tree *load = tree_child_cr(meta, s_load);

list_for_each(xfm, load->children)  {
  if (filter_matches(xfm, file)) { /* Needs to be added to transform.h */
    transform_load(aug, xfm, file);
    break;
  } 
}
```